### PR TITLE
Preparing for deltalake v0.17

### PR DESCRIPTION
**Pull request by rtyler**
Lots of refactorings necessary, the dynamodb lock mechanism will be changing too!

### CODE REVIEW

**Comment by rtyler** [comment]
I am building other branches off of this one to help do integration testing and qualification of the next `deltalake` crate release

File: src/lib.rs [hunk 1]
```diff
@@ -754,7 +754,7 @@ impl IngestProcessor {
         let dlq = dead_letter_queue_from_options(&opts).await?;
         let transformer = Transformer::from_transforms(&opts.transforms)?;
         let table = delta_helpers::load_table(table_uri, HashMap::new()).await?;
-        let coercion_tree = coercions::create_coercion_tree(&table.get_metadata()?.schema);
+        let coercion_tree = coercions::create_coercion_tree(table.schema().unwrap());
```
Comment (mightyshazam) [inline src/lib.rs:757]: Why switch from `?` to `unwrap`? While this _shouldn't_ really be a candidate for panicking, it is a possibility. Is that what we want?

File: src/lib.rs [hunk 2]
```diff
@@ -947,11 +947,11 @@ impl IngestProcessor {
 
         if self
             .delta_writer
-            .update_schema(self.table.get_metadata()?)?
+            .update_schema(self.table.state.delta_metadata().unwrap())?
```
Comment (mightyshazam) [inline src/lib.rs:950]: Same as above.

File: src/lib.rs [hunk 3]
```diff
@@ -947,11 +947,11 @@ impl IngestProcessor {
...
         {
             info!("Table schema has been updated");
             // Update the coercion tree to reflect the new schema
-            let coercion_tree = coercions::create_coercion_tree(&self.table.get_metadata()?.schema);
+            let coercion_tree = coercions::create_coercion_tree(self.table.schema().unwrap());
```
Comment (mightyshazam) [inline src/lib.rs:954]: Same as above.

**Review by mightyshazam — APPROVED** [review]
Looks good. I see that the delta crate has changed some things from results to options. That explains the `unwrap` calls replacing the `?` operations. We should consider addressing it at some point.

**Pull request merged**